pkg/join/join.go

```diff
@@ -0,0 +1,554 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package join
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/pkg/errors"
+	submariner "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/internal/constants"
+	"github.com/submariner-io/submariner-operator/internal/image"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/broker"
+	submarinerclientset "github.com/submariner-io/submariner-operator/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner-operator/pkg/discovery/globalnet"
+	"github.com/submariner-io/submariner-operator/pkg/discovery/network"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/datafile"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/brokersecret"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/servicediscoverycr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinercr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinerop"
+	"github.com/submariner-io/submariner-operator/pkg/version"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type Options struct {
+	ClusterID                     string
+	ServiceCIDR                   string
+	ClusterCIDR                   string
+	GlobalnetCIDR                 string
+	Repository                    string
+	ImageVersion                  string
+	NattPort                      int
+	IkePort                       int
+	PreferredServer               bool
+	ForceUDPEncaps                bool
+	ColorCodes                    string
+	NatTraversal                  bool
+	IgnoreRequirements            bool
+	GlobalnetEnabled              bool
+	IpsecDebug                    bool
+	SubmarinerDebug               bool
+	OperatorDebug                 bool
+	LabelGateway                  bool
+	LoadBalancerEnabled           bool
+	CableDriver                   string
+	GlobalnetClusterSize          uint
+	CustomDomains                 []string
+	ImageOverrideArr              []string
+	HealthCheckEnable             bool
+	HealthCheckInterval           uint64
+	HealthCheckMaxPacketLossCount uint64
+	CorednsCustomConfigMap        string
+}
+
+var status = cli.NewStatus()
+
+func SubmarinerCluster(subctlData *datafile.SubctlData, jo Options) error{
```

Comment:
Let's use the new structure introduced by https://github.com/submariner-io/submariner-operator/pull/1717 once it's merged.

```suggestion
func SubmarinerCluster(brokerInfo broker.Info, options Options) error{
```